### PR TITLE
Assignment 3   testing

### PR DESCRIPTION
**Pull request by carolinafm216**

### CODE REVIEW

File: assignments/life_expectancy/tests/test_cleaning_data_new.py
Comment (margarida-borbinha) [inline assignments/life_expectancy/tests/test_cleaning_data_new.py:1]: Avoid calling files *_new.py

File: assignments/life_expectancy/tests/test_cleaning.py [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
 
 
-def test_clean_data(pt_life_expectancy_expected):
+@patch("life_expectancy.loading_saving_data.save_data")
+@patch("life_expectancy.loading_saving_data.load_data")
+def test_clean_data(
+    mock_load, mock_save, life_expectancy_original, pt_life_expectancy_expected
+):
     """Run the `clean_data` function and compare the output to the expected output"""
-    #clean_data()
```
Comment (margarida-borbinha) [inline assignments/life_expectancy/tests/test_cleaning.py:10]: commented code

File: assignments/life_expectancy/tests/test_cleaning_data_new.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import pandas as pd
+from life_expectancy.data_cleaning import unpivot, convert_date_format, filter_region
+
+def test_unipivot(life_expectancy_unpivot, life_expectancy_original):
```
Comment (margarida-borbinha) [inline assignments/life_expectancy/tests/test_cleaning_data_new.py:4]: Having unpivot as a private function saves you of having to test it

File: assignments/life_expectancy/tests/test_cleaning_data_new.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    response_df = unpivot(life_expectancy_original)
+    # in order to force json file columns to be str
+    life_expectancy_unpivot = life_expectancy_unpivot.astype({"year": "str"})
+    life_expectancy_unpivot = life_expectancy_unpivot.astype({"value": "str"})
+    response_df = response_df.astype({"value": "str"})
+
+    pd.testing.assert_frame_equal(response_df, life_expectancy_unpivot)
+
+
+def test_convert_date_format(
```
Comment (margarida-borbinha) [inline assignments/life_expectancy/tests/test_cleaning_data_new.py:14]: Having convert_date as a private function saves you of having to test it

File: assignments/life_expectancy/tests/test_cleaning_data_new.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        {"year": "int"}
+    )
+    pd.testing.assert_frame_equal(
+        response_df.reset_index(drop=True),
+        life_expectancy_convert_date_format.reset_index(drop=True),
+        check_names=False,
+    )
+
+
+def test_filter_region(
```
Comment (margarida-borbinha) [inline assignments/life_expectancy/tests/test_cleaning_data_new.py:29]: Having filter_region as a private function saves you of having to test it

File: assignments/life_expectancy/tests/test_cleaning.py [hunk 5]
```diff
@@ -1,17 +1,23 @@
...
-    clean_data(OUTPUT_DIR / "eu_life_expectancy_raw.tsv",'PT')
-    pt_life_expectancy_actual = pd.read_csv(
-        OUTPUT_DIR / "pt_life_expectancy.csv"
-    )
+    mock_load.return_value = life_expectancy_original
+    mock_save.side_effect = print()
```
Comment (margarida-borbinha) [inline assignments/life_expectancy/tests/test_cleaning.py:16]: normally you use print when you _actually_ want to print something

**Review by margarida-borbinha — CHANGES_REQUESTED** [review]

File: assignments/life_expectancy/tests/test_cleaning.py (hunk 5 above)
Comment (MrCordeiro) [reply assignments/life_expectancy/tests/test_cleaning.py:16]: @margarida-borbinha given the requirements of the assignment, why don't you make a suggestion?

In general you should strike an appropriate balance between pointing out problems and providing direct guidance

File: assignments/life_expectancy/loading_saving_data.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import pathlib
+import pandas as pd
+
+read_file_name = "eu_life_expectancy_raw.tsv"
+file_path = pathlib.Path(__file__).parent / "data" / read_file_name
+
+
+def load_data(path: pathlib.Path) -> pd.DataFrame:
+    # with open(path, encoding="utf8") as file:
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/loading_saving_data.py:9]: ok

File: assignments/life_expectancy/tests/pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,33 @@
+[project]
+name = "assignments"
+version = "0.0.1"
+readme = "README.md"
+requires-python = ">=3.8"
+description = "First assignment of the FAAST Advance Foundations"
+authors = [
+    {name = "Carolina arreiros"}
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/tests/pyproject.toml:8]: ok

File: assignments/life_expectancy/tests/test_load_save_data_new.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+        raw_actual = load_data(file_path)
+
+        pd.testing.assert_frame_equal(raw_actual, life_expectancy_filter_region)
+        mock_read_csv.assert_called_once_with(file_path, sep="\t")
+
+
+def test_save_data(capfd):
+    msg = "I was saved"
+    # print (life_expectancy_mock)
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/tests/test_load_save_data_new.py:24]: ok

File: assignments/life_expectancy/tests/fixtures/life_expectancy_original.json [hunk 9]
```diff
@@ -0,0 +1,22 @@
+[
```
[the remark this answers is not in the document]
Comment (carolinafm216) [reply assignments/life_expectancy/tests/fixtures/life_expectancy_original.json:1]: ok

File: assignments/life_expectancy/tests/test_cleaning.py (hunk 1 above)
Comment (carolinafm216) [reply assignments/life_expectancy/tests/test_cleaning.py:10]: ok

File: assignments/life_expectancy/tests/test_cleaning_data_new.py (hunk 2 above)
Comment (carolinafm216) [reply assignments/life_expectancy/tests/test_cleaning_data_new.py:4]: ok

File: assignments/life_expectancy/tests/test_cleaning_data_new.py (hunk 3 above)
Comment (carolinafm216) [reply assignments/life_expectancy/tests/test_cleaning_data_new.py:14]: ok

File: assignments/life_expectancy/tests/test_cleaning_data_new.py (hunk 4 above)
Comment (carolinafm216) [reply assignments/life_expectancy/tests/test_cleaning_data_new.py:29]: ok

File: assignments/life_expectancy/tests/test_cleaning.py (hunk 5 above)
Comment (carolinafm216) [reply assignments/life_expectancy/tests/test_cleaning.py:16]: ok

**Pull request merged**